### PR TITLE
[yara] update Yara from v3.10.0 to v4.0.2

### PR DESCRIPTION
**Pull request by N3mes1s**
This will update the current port for Yara (v3.10.0) to support v4.0.2

### CODE REVIEW

File: ports/yara/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: yara
-Version: 3.10.0-1
+Version: 4.0.2-0
```
Comment (ras0219-msft) [inline ports/yara/CONTROL:2]:
```suggestion
Version: 4.0.2
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM once the Version field is changed.

It would be nice to use GLOBs in the buildsystem replacement to avoid needing these changes on future updates, but this is fine for now.

**Comment by NancyLi1013** [comment]
Hi @N3mes1s 

Would you like to update CMakeLists.txt as the suggestion from @ras0219-msft ?

**Comment by N3mes1s** [comment]
Sorry i m missing something here! what do you need me to do?

**Comment by NancyLi1013** [comment]
> Sorry i m missing something here! what do you need me to do?

Use GLOBs to set the source files that are needed to be compiled.

`file(GLOB/GLOB_RECURSE ...)`

> LGTM once the Version field is changed.
> 
> It would be nice to use GLOBs in the buildsystem replacement to avoid needing these changes on future updates, but this is fine for now.

Of course, if you don't, this is still ok.

**Comment by MFMokbel** [comment]
This issue is pending an author response as per the label "requires:author-response". @N3mes1s is this is something you can do? Perhaps the author of Yara has to do that change at the repo level. @NancyLi1013 can you carry on without that change request?

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@MFMokbel 
Thanks for your kindly reminder.

I merge master to retrigger CI. It seems good now. 
Let's wait for the CI test results.

**Comment by N3mes1s** [comment]
Yes, sorry about that. I will open a PR with the requested changes later on.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Comment by MFMokbel** [comment]
Thanks all!

**Pull request merged**